src/cashlib/cashlib.cpp

```diff
@@ -1,26 +1,41 @@
 // Copyright (c) 2015-2019 The Bitcoin Unlimited developers
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
-
 /* clang-format off */
 // must be first for windows
 #include "compat.h"
-
 /* clang-format on */
+
+#ifdef ANDROID // Workaround to fix gradle build
+#define SECP256K1_INLINE inline
+#endif
+
+#include "arith_uint256.h"
 #include "base58.h"
+#include "bloom.h"
+#include "cashaddrenc.h"
+#include "chainparams.h"
 #include "random.h"
+#include "script/sign.h"
 #include "streams.h"
 #include "uint256.h"
 #include "util.h"
 #include "utilstrencodings.h"
 
+#define MAX_SIG_LEN 100 // DER-encoded ECDSA is more like 72 but better to be safe
+
+#ifndef ANDROID
 #include <openssl/rand.h>
+#endif
+
```

Comment:
There should be some setup instructions in /doc/build-aux for setting up the build environment for android

ie.  sudo apt-get install default-jre,  sudo apt-get install default-jdk
and setting the JAVA_HOME env var etc...